src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -2589,8 +2559,11 @@ PSParallelCompact::update_and_deadwood_in_dense_prefix(ParCompactionManager* cm,
     const RegionData* const cp = sd.region(beg_region);
     if (cp->partial_obj_size() != 0) {
       beg_addr = sd.partial_obj_end(beg_region);
-    } else if (dead_space_crosses_boundary(cp, mbm->addr_to_bit(beg_addr))) {
-      beg_addr = mbm->find_obj_beg(beg_addr, end_addr);
+    } else {
+      idx_t beg_bit = mbm->addr_to_bit(beg_addr);
+      if (!mbm->is_obj_beg(beg_bit) && !mbm->is_obj_end(beg_bit - 1)) {
+        beg_addr = mbm->find_obj_beg(beg_addr, end_addr);
+      }
```

Comment:
I am not sure this refactoring improves upon the previous code. It adds an additional block level in the code, and removes the descriptive method name.
Actually this and the previous hunk is simply inlining of `dead_space_crosses_boundary`.

Not insisting on reverting this change, but I prefer putting more complicated conditions into descriptive helpers.

Reply:
I had great difficulty in understanding what `dead_space_crosses_boundary` does, so I inlined it to avoid the redundant partial-obj-size comparison  here -- it also makes the obj-bit checking more transparent.